src/types.ts

```diff
@@ -1,165 +1,153 @@
+	Which migration method to execute.
+	*/
+	readonly method: 'up' | 'down';
+}
+
+export interface UmzugMigrationParameters {
+	/**
+	The parameters that will be used to call the `up` and `down` migration functions.
+
+	It can be either an array (which defines the parameters directly), or a function (that will be called right before the migration execution) whose return value is the array of parameters.
+
+	@default []
+	*/
+	params?: any[] | (() => any[]);
+}
+
+export interface UmzugConstructorMigrationOptionsA extends ShortMigrationOptions, UmzugMigrationParameters {
+	/**
+	The path to the migrations directory, absolute or relative to `process.cwd()`.
+
+	Umzug will automatically retrieve migrations from this directory.
+
+	@default path.resolve(process.cwd(), 'migrations')
+	*/
+	readonly path?: string;
+
+	/**
+	A Regular Expression which determines whether or not a file found in the given path is a migration.
+
+	@default /^\d+[\w-]+\.js$/
+	*/
+	readonly pattern?: RegExp;
+
+	/**
+	Recursively search for migrations in the given folder.
+
+	@default false
+	*/
+	readonly traverseDirectories?: boolean;
+}
+
+export interface UmzugConstructorMigrationOptionsB extends Array<Migration>, UmzugMigrationParameters {}
+
+export type UmzugConstructorMigrationOptions = UmzugConstructorMigrationOptionsA | UmzugConstructorMigrationOptionsB;
+
+export interface UmzugConstructorOptions {
+	/**
+	The storage.
+
+	Possible values for built-in storages: 'none', 'json', 'mongodb', 'sequelize'.
+
+	You can also provide your own custom storage object. It must adhere to the UmzugStorage interface.
+
+	You can also provide a string which will be used as an argument for `require()`, which should require a class which implements the UmzugStorage interface.
+	*/
+	readonly storage?: string | UmzugStorage;
+
+	/**
+	The logging function.
+
+	A function that gets executed everytime migrations start and have ended.
+
+	Set to `false` to disable.
+	*/
+	readonly logging?: ((...args: any[]) => void) | false;
+
+	/**
+	The options for instantiating the storage. Check the available storages for further details.
+	*/
+	readonly storageOptions?: any;
+
+	/**
+	The options that will setup the detection of migrations.
+
+	This can be either a direct array of Migrations or a set of options which define the auto detection of migrations from a folder.
+	*/
+	readonly migrations?: UmzugConstructorMigrationOptions;
+}
+
+export type UmzugEventNames = 'migrating' | 'reverting' | 'migrated' | 'reverted';
+
+export type UmzugRunOptions =
+	/** Migration to execute. */
+	| string
+
+	/** Migrations to execute. */
+	| string[]
+
+	/** Object specifying which migrations to execute. */
+	| {
+			/** List of migrations to execute (by name). */
+			migrations?: string[];
+
+			/** The migration to start executing from. The given migration is not executed itself, only the ones after it. */
+			from?: string;
+
+			/** The last migration to execute (included). */
+			to?: string;
+	  };
```

Comment:
There is something odd with indentation here. Also last line has a tab followed by 2 spaces, is mixing tabs and spaces intentional?

Reply:
This is just prettier auto-formatting. In general, it doesn't allow you to decide on whitespace at all. I believe the reasoning for this mixed space/tab thing is because prettier is trying to align the closing brace with the opening one. The opening one has an extra indentation of two characters because of the `| `.